managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/NonDurableCursorImpl.java

```diff
@@ -40,7 +40,8 @@
         // Compare with "latest" position marker by using only the ledger id. Since the C++ client is using 48bits to
         // store the entryId, it's not able to pass a Long.max() as entryId. In this case there's no point to require
         // both ledgerId and entryId to be Long.max()
-        if (startCursorPosition == null || startCursorPosition.getLedgerId() == PositionImpl.latest.getLedgerId()) {
+        if (startCursorPosition == null || startCursorPosition.getLedgerId() == PositionImpl.latest.getLedgerId()
+                || startCursorPosition.compareTo(ledger.lastConfirmedEntry) > 0) {
```

Comment:
It seems that the condition `startCursorPosition.compareTo(ledger.lastConfirmedEntry) > 0` already cover the condition `startCursorPosition.getLedgerId() == PositionImpl.latest.getLedgerId()`, so compare with `ledger.lastConfirmedEntry` is enough?

Reply:
good suggestion.

Reply:
Do we have test case coverage?